cmd/buildkitd/main.go

```diff
@@ -316,6 +316,13 @@ func main() {
 			os.RemoveAll(lockPath)
 		}()
 
+		// listeners have to be initialized before the controller
```

Comment:
(This is because `newGRPCListeners()` is responsible for mkdiring and chowning `/run/buildkit`.)